proof/refine/AARCH64/ArchStateRelation.thy

```diff
@@ -427,78 +149,9 @@ definition arch_state_relation :: "(arch_state \<times> AARCH64_H.kernel_state)
-    \<rbrakk> \<Longrightarrow> R"
-  by (force simp: obj_relation_cuts_def2 is_other_obj_relation_type_def a_type_def
-                  cte_relation_def pte_relation_def tcb_relation_cut_def
-            split: Structures_A.kernel_object.splits kernel_object.splits if_splits
-                   AARCH64_A.arch_kernel_obj.splits)
-
-lemma eq_trans_helper:
-  "\<lbrakk> x = y; P y = Q \<rbrakk> \<Longrightarrow> P x = Q"
-  by simp
-
-lemma cap_relation_case':
-  "cap_relation cap cap' = (case cap of
-                              cap.ArchObjectCap arch_cap.ASIDControlCap \<Rightarrow> cap_relation cap cap'
-                            | _ \<Rightarrow> cap_relation cap cap')"
-  by (simp split: cap.split arch_cap.split)
-
-schematic_goal cap_relation_case:
-  "cap_relation cap cap' = ?P"
-  apply (subst cap_relation_case')
-  apply (clarsimp cong: cap.case_cong arch_cap.case_cong)
-  apply (rule refl)
-  done
-
-lemmas cap_relation_split =
-  eq_trans_helper [where P=P, OF cap_relation_case cap.split[where P=P]] for P
-lemmas cap_relation_split_asm =
-  eq_trans_helper [where P=P, OF cap_relation_case cap.split_asm[where P=P]] for P
-
-
-
-text \<open>
-  Relations on other data types that aren't stored but used as intermediate values
-  in the specs.
-\<close>
-primrec message_info_map :: "Structures_A.message_info \<Rightarrow> Types_H.message_info" where
-  "message_info_map (Structures_A.MI a b c d) = (Types_H.MI a b c d)"
-
-lemma mi_map_label[simp]: "msgLabel (message_info_map mi) = mi_label mi"
-  by (cases mi, simp)
-
-primrec syscall_error_map :: "ExceptionTypes_A.syscall_error \<Rightarrow> Fault_H.syscall_error" where
-  "syscall_error_map (ExceptionTypes_A.InvalidArgument n)   = Fault_H.InvalidArgument n"
-| "syscall_error_map (ExceptionTypes_A.InvalidCapability n) = (Fault_H.InvalidCapability n)"
-| "syscall_error_map ExceptionTypes_A.IllegalOperation      = Fault_H.IllegalOperation"
-| "syscall_error_map (ExceptionTypes_A.RangeError n m)      = Fault_H.RangeError n m"
-| "syscall_error_map ExceptionTypes_A.AlignmentError        = Fault_H.AlignmentError"
-| "syscall_error_map (ExceptionTypes_A.FailedLookup b lf)   = Fault_H.FailedLookup b (lookup_failure_map lf)"
-| "syscall_error_map ExceptionTypes_A.TruncatedMessage      = Fault_H.TruncatedMessage"
-| "syscall_error_map ExceptionTypes_A.DeleteFirst           = Fault_H.DeleteFirst"
-| "syscall_error_map ExceptionTypes_A.RevokeFirst           = Fault_H.RevokeFirst"
-| "syscall_error_map (ExceptionTypes_A.NotEnoughMemory n)   = Fault_H.syscall_error.NotEnoughMemory n"
-
+(* FIXME arch-split: object_type is arch-specific with one generic field, inverse of usual setup,
+   but if we split off the generic part, it would have to go before arch definitions, which would
+   require more locales or an extra generic theory file *)
 definition APIType_map :: "Structures_A.apiobject_type \<Rightarrow> AARCH64_H.object_type" where
```

Comment:
I wouldn't be opposed to a small extra theory if it makes the definitions nicer. We don't want that everywhere in the proofs, but for invariants and state relation I'd say it's worth it.

Reply:
Do we want to rename `APIType_map2` and move it here (or to a generic file if it can be split)? That could probably be done in a separate PR, except that we would lose track of this location and the connection to the definition that you're removing.

Reply:
Agree, I would do that when we hit Retype_R.

Reply:
When we get to Retype_R, we'll be forced to arch-split APIType_map2, which would be a good point to rename it. Given the grief it gave me today, I'm not likely to forget it.